lxd/container_lxc.go

```diff
@@ -1402,24 +1486,54 @@ func (c *containerLXC) expandConfig() error {
 func (c *containerLXC) expandDevices() error {
 	devices := types.Devices{}
 
+	var storagePoolsInProfile []string
 	// Apply all the profiles
 	for _, p := range c.profiles {
 		profileDevices, err := dbDevices(c.daemon.db, p, true)
 		if err != nil {
 			return err
 		}
 
+		// COMMENT(brauner): Check all pools specified in the
+		// container's profiles.
 		for k, v := range profileDevices {
+			if v["type"] == "disk" && v["path"] == "/" {
+				if v["pool"] != "" && !shared.StringInSlice(v["pool"], storagePoolsInProfile) {
+					storagePoolsInProfile = append(storagePoolsInProfile, v["pool"])
+				}
+			}
 			devices[k] = v
 		}
 	}
 
+	// COMMENT(brauner): Check if only one storage pool, the container is
+	// supposed to be created on is specified in its profiles.
+	if len(storagePoolsInProfile) > 1 {
+		return fmt.Errorf("Profiles the container belongs to specify different storage pools.")
+	}
```

Comment:
Profiles sure can specify root devices, I in fact have the root device in my "default" profile on all my production servers so I can alter the "size" for all containers in one go.

I really don't see the problem with having a chain of profiles that specify varying values of "pool" for the same device name, we only care about what the final value is, not how we ended up there.